ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -22,91 +22,95 @@
  */
 
 class IndexValue {
-  public enum Flags {
+  enum Flags {
     Delete_Index
   }
 
   private static int Blob_Size_In_Bytes = 8;
   private static int Offset_Size_In_Bytes = 8;
   private static int Flag_Size_In_Bytes = 1;
-  private static int Time_To_Live_Size_In_Bytes = 8;
+  private static int Expires_At_Ms_Size_In_Bytes = 8;
   private static int Original_Message_Offset_Size_In_Bytes = 8;
 
-  public static int Index_Value_Size_In_Bytes =
-      Blob_Size_In_Bytes + Offset_Size_In_Bytes + Flag_Size_In_Bytes + Time_To_Live_Size_In_Bytes
+  static int Index_Value_Size_In_Bytes =
+      Blob_Size_In_Bytes + Offset_Size_In_Bytes + Flag_Size_In_Bytes + Expires_At_Ms_Size_In_Bytes
           + Original_Message_Offset_Size_In_Bytes;
 
-  private ByteBuffer value;
+  private final ByteBuffer value;
+  private Offset offset;
 
-  public IndexValue(ByteBuffer value) {
+  IndexValue(String logSegmentName, ByteBuffer value) {
     if (value.capacity() != Index_Value_Size_In_Bytes) {
       throw new IllegalArgumentException("Invalid buffer size");
     }
     this.value = value;
+    offset = new Offset(logSegmentName, value.getLong(Blob_Size_In_Bytes));
   }
 
-  public IndexValue(long size, long offset, byte flags, long timeToLiveInMs) {
-    this(size, offset, flags, timeToLiveInMs, offset);
+  IndexValue(long size, Offset offset, byte flags, long expiresAtMs) {
+    this(size, offset, flags, expiresAtMs, offset.getOffset());
   }
 
-  public IndexValue(long size, long offset, byte flags, long timeToLiveInMs, long originalMessageOffset) {
+  IndexValue(long size, Offset offset, long expiresAtMs) {
+    this(size, offset, (byte) 0, expiresAtMs);
+  }
+
+  IndexValue(long size, Offset offset) {
+    this(size, offset, (byte) 0, Utils.Infinite_Time);
+  }
+
+  private IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long originalMessageOffset) {
+    this.offset = offset;
     value = ByteBuffer.allocate(Index_Value_Size_In_Bytes);
     value.putLong(size);
-    value.putLong(offset);
+    value.putLong(offset.getOffset());
     value.put(flags);
-    value.putLong(timeToLiveInMs);
+    value.putLong(expiresAtMs);
     value.putLong(originalMessageOffset);
     value.position(0);
```

Comment:
No idea - Don't see the reason to do it either way and since this is not a part of my changes, will leave as is.